### DISCUSSION TITLE
Purge/Delete the historical data from pgwatch(db: base)

### DISCUSSION THREAD

**Discussion by Sarang-Dharme-od** [cybertec-postgresql/pgwatch2#600] [Q&A]
We are running pgwatch container using _cybertec/pgwatch2-postgres:1.9.0_ docker image. We has been storing/persisting postgres data in docker volume(named pg) on host (-v pg:/var/lib/postgresql). At the moment our disk is full(100%) due to database _base_. Is there any way I can purge/delete the historical data from pgwatch to release disk space on my server?

![image](https://user-images.githubusercontent.com/53090010/216335788-0ee44a79-1590-41cf-bcf4-4a6fbcc66998.png)

**Comment by pashagolub** [comment]
Hello, 

yeah, sure! You can easily delete old data if you don't need it. Please, check the [Storage Monitoring]( https://pgwatch2.readthedocs.io/en/latest/long_term_installations.html#storage-monitoring) section of our official manual for the hints!

Kind regards

**Reply by Sarang-Dharme-od** [reply]
Hello @pashagolub 

As per the reference document [Storage Monitoring](https://pgwatch2.readthedocs.io/en/latest/long_term_installations.html#storage-monitoring), I have set PW2_PG_RETENTION_DAYS enviroment variable to 4 days. However, It's not working for me. No effect on retention. Can you please advice?

code Snippet:
services:
  pgwatch:
    container_name: pgwatch2
    image: "cybertec/pgwatch2-postgres:1.9.0"
    environment:
      PW2_TESTDB: "true"
      PW2_DATASTORE: "postgres"
      PW2_PG_RETENTION_DAYS: 4

**Reply by pashagolub** [reply]
Can you please elaborate more on "not working"? Some tables are growing? What are they? Settings are not applied?

Also please consider our [Producst support](https://www.cybertec-postgresql.com/en/services/postgresql-support/standard/product-support/) service.

**Reply by Sarang-Dharme-od** [reply]
@pashagolub  I have set PW2_PG_RETENTION_DAYS enviroment variable to 4 days. Ideally data older than 4 days should not be visible in dashboard as it should be purged/deleted. However, Still data is being persisted for 15-16 days. I have attached screenshot of the dashboard.
![image](https://user-images.githubusercontent.com/53090010/224006254-f2bc2db8-244c-48da-8544-e42b05af41a3.png)

**Reply by pashagolub** [reply]
Aha, thanks for the clarification. According to [the logic](https://github.com/cybertec-postgresql/pgwatch2/blob/4a15bfe79276aa06e414a499f37be86a4a070a89/pgwatch2/pgwatch2.go#L1298-L1425) old metrics never deleted at once, because this may kill the server. Instead it deleted in chunks.

If you're sure you need to delete everything, please, find the proper SQL following the link provided.

**Comment by sgnatjuk** [comment]
Hi!
Did the problem come up?
I have the same question. Can you help you information?
I asked a similar question:
https://github.com/cybertec-postgresql/pgwatch2/discussions/831
Thanks
